### PR TITLE
drag-drop items to turf to sort by type & move them

### PR DESCRIPTION
**Pull request by Delimusca**
this lets you dragdrop items to turfs, which will move them to that turf along with any items of the same type.
good for sorting mineral bars, moving lots of paper, etc.

### CODE REVIEW

File: code/game/turfs/simulated/floor.dm [hunk 1]
```diff
@@ -182,4 +182,13 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
...
 					else if(dirt > 100)
-						dirtoverlay.alpha = min(dirtoverlay.alpha+10, 200)
\ No newline at end of file
+						dirtoverlay.alpha = min(dirtoverlay.alpha+10, 200)
+
+/turf/simulated/floor/MouseDrop_T(atom/dropping, mob/user)
+	if(istype(dropping, /obj/item) && istype(dropping.loc, /turf))
+		var/obj/item/I = dropping
+		if(isliving(user) && !user.stat)
+			for(I in I.loc)
```
Comment (RemieRichards) [inline code/game/turfs/simulated/floor.dm]: not entirely certain but this smells like you could pull items OUT of boxes/containers erroneously.

**Comment by RemieRichards** [comment]
Interesting concept, the exact same can be achieved with obj/item/MouseDrop() and an istype for the turf but both are just mirrors of eachother so it's no issue.

File: code/game/turfs/simulated/floor.dm (hunk 1 above)
Comment (Delimusca) [inline code/game/turfs/simulated/floor.dm]: i noticed that problem, putting istype(dropping.loc, /turf) fixed that.

Comment (RemieRichards) [inline code/game/turfs/simulated/floor.dm]: *Doh!* I am an idiot, Nice to see you solved it though.

**Comment by Alek2ander** [comment]
You should check for anchored on every item, just in case

**Comment by Delimusca** [comment]
added anchored check, added a sleep(1) to make sorting look nice.

**Comment by paprka** [comment]
This fucking owns.

**Comment by Razharas** [comment]
You sure there is check for range?
Or we will have telecinetic sorting from distances

**Comment by Delimusca** [comment]
mousedrop_t should only be called when the user is adjacent to the item and the target, according to [drag_drop.dm](https://github.com/tgstation/-tg-station/blob/master/code/_onclick/drag_drop.dm). last time i tested it, this seemed to be correct.

**Pull request closed without merging**